packages/typespec-go/src/tcgcadapter/clients.ts

```diff
@@ -223,6 +223,20 @@ export class clientAdapter {
     }
   }
 
+  private adaptContentType(contentTypeStr: string): 'binary' | 'JSON' | 'Text' | 'XML' {
+    // we only recognize/support JSON, text, and XML content types, so assume anything else is binary
+    // TODO: this seems goofy but maybe it's ok?
```

Comment:
Is the goofy part that we assume binary?

It does seem odd that contentType isn't required. Is this by chance because the http spec says by default its binary?

Reply:
I just don't like having a heuristic like this. It's probably fine in practice for Azure though. I'll remove the `TODO` and if it comes back to bite us later we can improve it then.

RFC 822 says the default is `text/plain`. I don't know if this has been superseded though.